src/js/actions/LibraryActions.js

```diff
@@ -97,85 +96,63 @@ export default {
     },
 
     refresh: function() {
-
-        const folders = app.config.get('musicFolders');
-
         AppDispatcher.dispatch({
             actionType : AppConstants.APP_LIBRARY_REFRESH_START
         });
 
-        // Start the big thing
-        app.db.remove({ type : 'track' }, { multi: true }, (err) => {
-            if(err) console.warn(err);
-            app.db.loadDatabase((err) => {
-                if(err) console.warn(err);
-                else {
-
-                    let filesList = [];
+        const dispatchEnd = function() {
+            AppDispatcher.dispatch({
+                actionType : AppConstants.APP_LIBRARY_REFRESH_END
+            });
+        };
+        const folders = app.config.get('musicFolders');
+        const supportedExtensionsGlob = `**/*.{${ app.supportedExtensions.join() }}`;
+        const fsConcurrency = 64;
 
-                    // Loop through folders
-                    folders.forEach((folder) => {
-                        // Get the list of files
-                        filesList = filesList.concat(walkSync(folder, { directories: false }).map((d) => path.join(folder, d)));
-                    });
+        const getMetadataAsync = function(track) {
+            return new Promise((resolve) => {
+                utils.getMetadata(track, resolve);
+            });
+        };
 
-                    // Get the metadatas of all the files
-                    const filesListFiltered = filesList.map((file) => {
-                        return {
-                            path: fs.realpathSync(file),
-                            mime: mime.lookup(file)
-                        };
-                    }).filter((track) => {
-                        return app.supportedFormats.includes(track.mime);
-                    });
+        // Start the big thing
+        app.db.removeAsync({ type : 'track' }, { multi: true }).then(() => {
```

Comment:
it came from [`Promise.promisifyAll`](http://bluebirdjs.com/docs/api/promise.promisifyall.html)